Improve FindCUDALibs.

joaander opened the pull request:
<!-- Please confirm that your work is based on the correct branch. -->
<!-- Bug fixes should be based on *trunk-patch*. -->
<!-- Backwards compatible new features should be based on *trunk-minor*. -->
<!-- Incompatible API changes should be based on *trunk-major*. -->

## Description

<!-- Describe your changes in detail. -->
* Add status messages for found cuda libraries.
* Remove optional libraries/executables from required list.

## Motivation and context

<!--- Why is this change required? What problem does it solve? -->
Starting with CUDA 12, conda-forge is now packaging libraries separately and does not package all libraries and tools in the cuda toolkit (https://github.com/conda-forge/hoomd-feedstock/pull/83). These changes make it easier to troubleshoot which libraries are missing and allows builds to proceed when optional libraries or executables are missing.

## How has this been tested?

<!--- Please describe in detail how you tested your changes. -->
Tested in https://github.com/conda-forge/hoomd-feedstock/pull/83 and locally.

<!--- Please build the sphinx documentation and check that any changes to
      documentation display properly. -->

## Change log

<!-- Propose a change log entry. -->
```
Added:

* More detailed status message for found CUDA libraries.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

joaander committed:
Improve status messages for found cuda libraries.

Also remove optional libraries/executables from required list.

b-butler commented on CMake/hoomd/FindCUDALibs.cmake:
```diff
@@ -49,6 +49,7 @@ if (HIP_PLATFORM STREQUAL "nvcc")
         INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}"
       )
     endif()
+    find_package_message(CUDALibs "Found nvrtc: ${CUDA_nvrtc_LIBRARY}" "[CUDA_nvrtc_LIBRARY]")
```
I assume that you are using CUDALibs rather than the specific library/program/package in these messages on puprose?

joaander replied:
On testing further I found that the first argument to `find_package_message` needs to be unique for each library to avoid repeated messages: https://cmake.org/cmake/help/latest/module/FindPackageMessage.html. See the fix in 64cbf00.

joaander committed:
Don't repeat found messages.

joaander committed:
Show python find messages again.

The repeated pybind11 found message is annoying. However, using QUIET
here hides the PythonInterp found message.... Perhaps we should post
an upstream fix to pybind11.
